packages/workflow/src/features/correction/routes.ts

```diff
@@ -293,28 +298,56 @@ export const routes = [
       const { correction, ...registration } =
         recordInputWithUploadedAttachments.registration!
 
+      /*
+       * Separate tasks from the record
+       */
+      const [correctionRequestApprovedTask, correctedTask] =
+        getTasksInAscendingOrder(recordInCorrectedState)
+      const recordWithoutTasks = getRecordWithoutTasks(recordInCorrectedState)
+
+      /*
+       * Create and update the valid record that only has one task (the corrected task)
+       */
+      const newRecord = addTaskToRecord(recordWithoutTasks, correctedTask)
+
+      newRecord satisfies ValidRecord
+
       const recordWithUpdatedValues = updateFHIRBundle(
-        recordInCorrectedState,
+        newRecord,
         { ...recordInputWithUploadedAttachments, registration },
         getEventType(record)
       )
 
       /*
        * Saves the previous task in the "approved" state and the new task (record put to Registered state)
+       * Note that because we are storing the task "twice", the approved task needs to be first in the list!
        */
 
-      await sendBundleToHearth(recordWithUpdatedValues)
-
-      const recordWithOnlyTheNewTask = withOnlyLatestTask(
+      const updatedCorrectedTask = getTaskFromSavedBundle(
         recordWithUpdatedValues
       )
+      const recordWithNoTasks = getRecordWithoutTasks(recordWithUpdatedValues)
+
+      // First add the approved task to the record
+      const recordWithApprovedTask = addTaskToRecord(
+        recordWithNoTasks,
+        correctionRequestApprovedTask
+      )
+
+      // Then add the corrected task to the record
+      const fullBundleToBeSaved = addTaskToRecord(
+        recordWithApprovedTask,
+        updatedCorrectedTask
+      )
```

Comment:
Here I once more remove all tasks from the bundle, and then add both the approved task and then the corrected task back in that specific order. This is so Hearth saves them in the same order.